Update vivado_structural_error_injection to only look for files it created this invocation

When you run a flow with structural comparison (`vivado_phys_netlist_cmp`), followed by the `vivado_structural_error_injection` flow, an exception occurs because the former flow leaves a file in the `struct_cmp` directory called `struct_cmp.log`.  This breaks the error injection code as it is not expecting this file to be here.

I've added a work-around [here](https://github.com/byuccl/bfasst/blob/831e41eb20cfa52ff44fd4b2ab044f4f5268ee18/bfasst/ninja_flows/vivado_structural_error_injection.py#L77-L78), but I wonder the larger implications of this?  What is the correct way to make sure we don't have these kinds of issues?

I tried the following runs which resulted in a different error than I expected:
```
python scripts/run.py vivado_phys_netlist_cmp byu/alu
>>> [11/11] structurally compare /home/keenanrf/research/bfasst/build/byu/alu/xray/alu_reversed.v /home/keenanrf/research/bfasst/build/byu/alu/vivado_phys_netlist/viv_impl_physical.v

python scripts/run.py vivado_structural_error_injection byu/alu
>>> [36/400] structurally compare /home/keenanrf/research/bfasst/build/byu/alu/vivado_phys_netlist/viv_impl_physical.v /home/keenanrf/research/bfasst/build/byu/alu/error_injection/bit_flip_12.v
FAILED: /home/keenanrf/research/bfasst/build/byu/alu/struct_cmp/bit_flip_12_cmp.log
python bfasst/ninja_utils/structural.py --build_dir /home/keenanrf/research/bfasst/build/byu/alu --netlists /home/keenanrf/research/bfasst/build/byu/alu/vivado_phys_netlist/viv_impl_physical.v /home/keenanrf/research/bfasst/build/byu/alu/error_injection/bit_flip_12.v --log_path /home/keenanrf/research/bfasst/build/byu/alu/struct_cmp/bit_flip_12_cmp.log
Traceback (most recent call last):
  File "/home/keenanrf/research/bfasst/bfasst/ninja_utils/structural.py", line 770, in <module>
    struct_cmp.compare_netlists()
  File "/home/keenanrf/research/bfasst/bfasst/ninja_utils/structural.py", line 171, in compare_netlists
    self.perform_mapping()
  File "/home/keenanrf/research/bfasst/bfasst/ninja_utils/structural.py", line 411, in perform_mapping
    self.potential_mapping_wrapper(next(instance_iter))
  File "/home/keenanrf/research/bfasst/bfasst/ninja_utils/structural.py", line 357, in potential_mapping_wrapper
    self.add_block_mapping(instance, matched_instance)
  File "/home/keenanrf/research/bfasst/bfasst/ninja_utils/structural.py", line 535, in add_block_mapping
    self.add_net_mapping(net_a, net_b)
  File "/home/keenanrf/research/bfasst/bfasst/ninja_utils/structural.py", line 548, in add_net_mapping
    raise AssertionError(
AssertionError: CLBLM_R_X35Y120_SLICE_X52Y120_DO6 in net_mapping.inverse already. net1: \result_OBUF[11]_inst_i_5_n_0
[45/400] inject a BIT_FLIP into xrev netlist for /home/keenanrf/research/bfasst/build/byu/alu
ninja: build stopped: subcommand failed.
Error injection failed on bit_flip_12
ERROR: Ninja failed with return code 1
```
@jgoeders is there a specific design you ran that triggered your exception? This exception seems unrelated to the presence of `struct_cmp.log`

--- a/bfasst/flows/vivado_structural_error_injection.py
+++ b/bfasst/flows/vivado_structural_error_injection.py
@@ -43,8 +43,7 @@ class VivadoStructuralErrorInjection(Flow):
         self.xrev_tool.create_build_snippets(self.vivado_impl_tool.outputs["bitstream"])
 
         random_seed_multiplier = 1
-        error_type = [ErrorType.BIT_FLIP, ErrorType.WIRE_SWAP]
-        for error in error_type:
+        for error in ErrorType:
             for i in range(1, self.num_runs + 1):
                 self.error_injector_tool.create_build_snippets(
                     error_type=error,
@@ -71,17 +70,22 @@ class VivadoStructuralErrorInjection(Flow):
         cmp_dir = self.compare_tool.build_path
         error_dir = self.error_injector_tool.build_path
 
-        for file in cmp_dir.iterdir():
-            if not file.name.startswith("bit_flip") and not file.name.startswith("wire_swap"):
-                continue
-            with open(file, "r") as f:
-                # SUCCESS means the compare tool did not detect an actual error
-                if "SUCCESS: Structural comparison found mismatch as expected" in f.read():
-                    err_log_name = file.name.split("_cmp.log")[0] + ".log"
-                    err_log = error_dir / err_log_name
-                    err_netlist = err_log.with_suffix(".v")
-                    file.unlink()
-                    err_log.unlink()
-                    err_netlist.unlink()
-                else:
-                    print(f"Error injection failed on {file.name.split('_cmp.log')[0]}")
+        for error in ErrorType:
+            for i in range(1, self.num_runs + 1):
+                # remove the error injection artifact, the corrupt netlist, and the compare log if
+                # the compare tool says the two netlists are different
+                basename = f"{error.name.lower()}_{i}"
+                cmp_log = cmp_dir / f"{basename}_cmp.log"
+                with open(cmp_log, "r") as f:
+                    if "SUCCESS: Structural comparison found mismatch as expected" in f.read():
+                        err_log = error_dir / f"{basename}.log"
+                        err_netlist = err_log.with_suffix(".v")
+                        cmp_log.unlink()
+                        err_log.unlink()
+                        err_netlist.unlink()
+                    else:
+                        raise CompareFalsePositiveError(f"Error injection failed on {basename}")
+
+
+class CompareFalsePositiveError(RuntimeError):
+    """Raised when a compare tool incorrectly compares two netlist as equivalent."""

